### ISSUE TITLE
Docstring revisions for Colout, CutOut, CutMix, Layer Freezing, Mixup, Label Smoothing, Progressive Resizing

### ISSUE THREAD

**Issue by ravi-mosaicml** [mosaicml/composer#542]
List of revisions for #483:

- [ ] https://mosaicml-composer--483.com.readthedocs.build/en/483/api_reference/composer.algorithms.colout.colout.html#module-composer.algorithms.colout.colout: In the ColOut examples for ColOutTransform and colout_batch, can you construct a simple training loop, showing where colout would be applied. I don't think this is needed for colout_image since one would never use that function directly in the training loop.
- [ ] https://mosaicml-composer--483.com.readthedocs.build/en/483/api_reference/composer.algorithms.cutout.cutout.html#module-composer.algorithms.cutout.cutout:
  - [x] Can the docstrings for `apply` and `match` be removed and instead, in the init signature, mention when it runs?
  - [ ] Can you also add a link back to the method card in the docstrings?
  - [ ] For `cutout_batch`, can you include a training loop example.
  - [x] Many parameters are missing type annotations.
- [ ] https://mosaicml-composer--483.com.readthedocs.build/en/483/api_reference/composer.algorithms.cutmix.cutmix.html#module-composer.algorithms.cutmix.cutmix
  - [ ] For cutmix, can you add links back to the method card in the docstrings?
  - [x] Same feedback for match and apply, as well as ensuring that each parameter has a type annotation.
  - [x] The cutmix_batch could also use formatting for the equations. For the return value, can you put it in one line with Tuple[torch.Tensor, torch.Tensor] notation (since functions only return one value)?
  - [x] "Both the original and shuffled labels are returned. This is done because for many loss functions (such as cross entropy) the targets are given as indices, so interpolation must be handled separately." -- could you elaborate what this means, perhaps with an example? I also don't think this is correct, as the return statement doesn't return the original labels -- just the new samples and new labels?
- [ ] [Layer freezing](https://mosaicml-composer--483.com.readthedocs.build/en/483/api_reference/composer.algorithms.layer_freezing.layer_freezing.html#module-composer.algorithms.layer_freezing.layer_freezing)
  - [ ] Link back to method card
  - [x] Move docstrings for apply, find_unused_parameters, and match to the `__init__` docstring
  - [x] The formatting for the functional example is very indented
  - [ ] Can you add a dummy training loop showing where the function would be called inside of it?
  - [ ] The return line is missing the type annotation, so sphinx added a return type line below it
- [ ] https://mosaicml-composer--483.com.readthedocs.build/en/483/api_reference/composer.algorithms.mixup.mixup.html#module-composer.algorithms.mixup.mixup: Same comments as in other algorithms. Also, "convex combinations of pairs of examples and targets" didn't make sense to me -- I had to look at the example image in the method card to understand what this meant. Would something like "Mixup creates synthetic samples and targets by averaging together pairs of samples in a batch and setting the targets to the proportion of each class in the averaged sample" (I'm probably off here with what is actually happening, but I had no idea what a convex combination was. Feel free to ignore if this term is common in the CV world.) 
- [ ] https://mosaicml-composer--483.com.readthedocs.build/en/483/api_reference/composer.algorithms.label_smoothing.label_smoothing.html#module-composer.algorithms.label_smoothing.label_smoothing: Same comments as in other algorithms. In addition could ` (1 - alpha) * targets + alpha * smoothed_targets` be in preformatted or math noation?
- [ ] https://mosaicml-composer--483.com.readthedocs.build/en/483/api_reference/composer.algorithms.progressive_resizing.progressive_resizing.html#composer.algorithms.progressive_resizing.progressive_resizing.ProgressiveResizing: Same comments as in other algorithms

**Comment by mvpatel2000** [comment]
Closing. Tracking elsewhere as low pri
